bin/install_swiftpm_dependencies

```diff
@@ -37,29 +98,34 @@ sudo defaults write com.apple.dt.Xcode IDEPackageSupportUseBuiltinSCM YES
 add_host_to_ssh_known_hosts bitbucket.org
 add_host_to_ssh_known_hosts github.com
 
-# Based on the project type, resolve the packages using the correct build type
-if [[ "$BUILD_TYPE" == "XCODEBUILD" ]]; then
-  echo "	Resolving packages with \`xcodebuild\`"
-  xcodebuild -resolvePackageDependencies -onlyUsePackageVersionsFromResolvedFile
-elif [[ "$BUILD_TYPE" == "SWIFT" ]]; then
-  echo "	Resolving packages with \`swift\`"
+# Resolve the packages using the correct method
+if [[ -d "${XCWORKSPACE_PATH}" ]]; then
+  echo "~~~ Resolving Swift Packages with \`xcodebuild\`"
+  if [[ -z "${SCHEME_NAME}" ]]; then
+    SCHEME_NAME=$(xcodebuild -workspace "${XCWORKSPACE_PATH}" -onlyUsePackageVersionsFromResolvedFile -skipPackageUpdates -list | sed -n '/Schemes:/{n;s/^ *//p;}')
+    echo "No \`--scheme\` parameter provided; assuming the first one found (as it shouldn't matter in practice?): \`${SCHEME_NAME}\`"
+  fi
+  echo "Using -workspace \"${XCWORKSPACE_PATH}\" -scheme \"${SCHEME_NAME}\""
+  xcodebuild -workspace "${XCWORKSPACE_PATH}" -scheme "${SCHEME_NAME}" -resolvePackageDependencies -onlyUsePackageVersionsFromResolvedFile
```

Comment:
💡 🤞   I'll have to do some more tests(†) to validate this, but from my quick experiments it seems that when I ran `xcodebuild -workspace "${XCWORKSPACE_PATH}" -list` (initially to guess a scheme to use)… `xcodebuild` printed "Resolve Package Graph", suggesting that it was resolving dependencies first before finally printing the list of schemes…

Which might mean that we could use this to our advantage, to just pass `-list` instead of having to pass an explicit `-scheme NAME` (i.e. `xcodebuild -workspace -resolvePackageDependencies -onlyUsePackageVersionsFromResolvedFile -list`)… which would completely avoid the need to pass a `-scheme` to the command in the first place (just because of the bug of `xcodebuild` requiring `-scheme` [or `-list`] to be provided when `-workspace` is also provided… even in cases where the scheme doesn't matter, like iinm in the case of resolving dependencies)

---

(†) we should test this theory thoroughly though, especially with some edge cases—like deliberately have an outdated `Package.resolved` pointing to older versions, and/or ensuring the packages are not present yet locally, before running those commands—to check especially if the `-onlyUsePackageVersionsFromResolvedFile` is honored by `xcodebuild` (as opposed to being ignored when we use `-list` instead of an explicit `-scheme NAME`).

---

**But if this ends up behaving like I hope, this means that we could simplify the whole code of this bash script by**:
 - Removing the `--scheme` input parameter parsing from this utility
 - Directly invoke `xcodebuild -workspace "${XCWORKSPACE_PATH}" -resolvePackageDependencies -onlyUsePackageVersionsFromResolvedFile -list`—in order to resolve the Swift Dependencies with `xcodebuild` while **still ensuring it'd use the `.xcworkspace`** and not `.xcodeproj`, yet **without having to provide a `-scheme`** option

Reply:
>  but from my quick experiments it seems that when I ran xcodebuild -workspace "${XCWORKSPACE_PATH}" -list (initially to guess a scheme to use)… xcodebuild printed "Resolve Package Graph", suggesting that it was resolving dependencies first before finally printing the list of schemes…

That's what I observe on my end, too. In Pocket Casts iOS:

```
➜ xcodebuild -workspace "podcasts.xcworkspace" -list
Command line invocation:
    /Applications/Xcode-16.0.0-Beta.3.app/Contents/Developer/usr/bin/xcodebuild -workspace podcasts.xcworkspace -list

User defaults from command line:
    IDEPackageSupportUseBuiltinSCM = YES

Resolve Package Graph


Resolved source packages:
  AutomatticTracksiOS: https://github.com/Automattic/Automattic-Tracks-iOS @ 3.4.2
  SwiftProtobuf: https://github.com/apple/swift-protobuf.git @ 1.22.1
  GTMAppAuth: https://github.com/google/GTMAppAuth.git @ 4.1.1
  Kingfisher: https://github.com/onevcat/Kingfisher @ 7.10.2
  TinyCSV: https://github.com/dagronf/TinyCSV @ 1.0.0
  JLRoutes: https://github.com/joeldev/JLRoutes @ 2.1.1
  Agrume: https://github.com/Automattic/Agrume @ 5.6.13
  Sentry: https://github.com/getsentry/sentry-cocoa @ 8.29.1
  SwiftSubtitles: https://github.com/dagronf/SwiftSubtitles @ 1.5.2
  DSFRegex: https://github.com/dagronf/DSFRegex @ 3.4.0
  Firebase: https://github.com/firebase/firebase-ios-sdk.git @ 10.25.0
  GoogleSignIn: https://github.com/google/GoogleSignIn-iOS @ 7.1.0
  abseil: https://github.com/google/abseil-cpp-binary.git @ 1.2024011601.1
  AppAuth: https://github.com/openid/AppAuth-iOS.git @ 1.7.5
  DifferenceKit: https://github.com/ra1028/DifferenceKit @ 1.2.0
  UIDeviceIdentifier: https://github.com/squarefrog/UIDeviceIdentifier @ 2.1.0
  AppCheck: https://github.com/google/app-check.git @ 10.19.0
  leveldb: https://github.com/firebase/leveldb.git @ 1.22.2
  GoogleAppMeasurement: https://github.com/google/GoogleAppMeasurement.git @ 10.25.0
  InteropForGoogle: https://github.com/google/interop-ios-for-google-sdks.git @ 100.0.0
  BuildkiteTestCollector: https://github.com/buildkite/test-collector-swift @ 0.1.1
  GoogleUtilities: https://github.com/google/GoogleUtilities.git @ 7.12.1
  Lottie: https://github.com/airbnb/lottie-ios.git @ 4.4.0
  Promises: https://github.com/google/promises.git @ 2.3.1
  PocketCastsServer: /Users/gio/Developer/a8c/pcios/Modules/Server
  GoogleDataTransport: https://github.com/google/GoogleDataTransport.git @ 9.3.0
  PocketCastsDataModel: /Users/gio/Developer/a8c/pcios/Modules/DataModel
  Sodium: https://github.com/jedisct1/swift-sodium @ 0.9.1
  gRPC: https://github.com/google/grpc-binary.git @ 1.62.2
  nanopb: https://github.com/firebase/nanopb.git @ 2.30909.0
  Fuse: https://github.com/krisk/fuse-swift @ 1.4.0
  FMDB: https://github.com/ccgus/fmdb.git @ 2.7.8
  GTMSessionFetcher: https://github.com/google/gtm-session-fetcher.git @ 3.4.1
  PocketCastsUtils: /Users/gio/Developer/a8c/pcios/Modules/Utils
  SwipeCellKit: https://github.com/shiftyjelly/SwipeCellKit @ 2.7.7

Information about workspace "podcasts":
    Schemes:
        BuildkiteTestCollector
        DSFRegex
        google-cast-sdk-no-bluetooth
        NotificationContent
        NotificationExtension
        Pocket Casts Prototype Build
        Pocket Casts Staging
        Pocket Casts Watch App
        pocketcasts
        PocketCastsDataModel
        PocketCastsServer
        PocketCastsUtils
        PodcastsIntents
        PodcastsIntentsUI
        Pods-PocketCastsTests
        Pods-podcasts
        PreBuildActions
        PulseCore
        PulseUI
        Screenshot Automation
        Screenshot Automation Watch
        Share Extension
        SwiftGen
        SwiftLint
        WidgetExtension
```